classes/class-wc-connect-nux.php

```diff
@@ -83,5 +97,288 @@ public function register_add_service_to_zone_pointer( $pointers ) {
 			);
 			return $pointers;
 		}
+
+		public function get_jetpack_install_status() {
+			// check if Jetpack is activated
+			if ( ! class_exists( 'Jetpack_Data' ) ) {
+				// not activated, check if installed
+				if ( 0 === validate_plugin( 'jetpack/jetpack.php' ) ) {
+					return self::JETPACK_INSTALLED;
+				}
+				return self::JETPACK_UNINSTALLED;
+			} else if ( defined( 'JETPACK_DEV_DEBUG' ) && true === JETPACK_DEV_DEBUG ) {
+				// installed, activated, and dev mode on
+				return self::JETPACK_DEV;
+			}
+
+			// installed, activated, dev mode off
+			// check if connected
+			$user_token = Jetpack_Data::get_access_token( JETPACK_MASTER_USER );
+			if ( isset( $user_token->external_user_id ) ) { // always an int
+				return self::JETPACK_CONNECTED;
+			}
+
+			return self::JETPACK_ACTIVATED;
+		}
+
+		public function should_display_nux_notice_on_screen( $screen ) {
+			if ( // Display if on any of these admin pages.
+				( // Products list.
+					'product' === $screen->post_type
+					&& 'edit' === $screen->base
+				)
+				|| ( // Orders list.
+					'shop_order' === $screen->post_type
+					&& 'edit' === $screen->base
+					)
+				|| ( // Edit order page.
+					'shop_order' === $screen->post_type
+					&& 'post' === $screen->base
+					)
+				|| ( // WooCommerce settings.
+					'woocommerce_page_wc-settings' === $screen->base
+					)
+				|| ( // WooCommerce featured extension page
+					'woocommerce_page_wc-addons' === $screen->base
+					&& isset( $_GET['section'] ) && 'featured' === $_GET['section']
+					)
+				|| ( // WooCommerce shipping extension page
+					'woocommerce_page_wc-addons' === $screen->base
+					&& isset( $_GET['section'] ) && 'shipping_methods' === $_GET['section']
+					)
+				|| 'plugins' === $screen->base
+			) {
+				return true;
+			}
+			return false;
+		}
+
+		public function should_display_nux_notice_for_current_store_locale() {
+			$base_location = wc_get_base_location();
+			$country = isset( $base_location['country'] )
+				? $base_location['country']
+				: '';
+			// Do not display for non-US, non-CA stores.
+			if( 'CA' === $country || 'US' === $country ) {
+				return true;
+			}
+			return false;
+		}
+
+		public function get_jetpack_redirect_url() {
+			$full_path = add_query_arg( array() );
+			// Remove [...]/wp-admin so we can use admin_url().
+			$new_index = strpos( '/wp-admin', $full_path ) + strlen( '/wp-admin' );
+			$path = substr( $full_path, $new_index );
+			return admin_url( $path );
+		}
+
+		public function set_up_nux_notices() {
+			if ( ! current_user_can( 'manage_woocommerce' )
+				|| ! current_user_can( 'install_plugins' )
+				|| ! current_user_can( 'activate_plugins' )
+			) {
+				return;
+			}
+
+			if ( ! $this->should_display_nux_notice_for_current_store_locale() ) {
+				return;
+			}
+
+			$jetpack_install_status = $this->get_jetpack_install_status();
+
+			$ajax_data = array(
```

Comment:
You're only using data in one part of a switch statement. Seems kind of wasteful to execute it in all situations. Moreover, it looks like this code will be executed on every page load, even after the NUX flow has been completed...